vitrivr-engine-module-fes/src/main/kotlin/org/vitrivr/engine/base/features/external/common/ApiWrapper.kt

```diff
@@ -0,0 +1,323 @@
+package org.vitrivr.engine.base.features.external.common
+
+import okhttp3.OkHttpClient
+import org.openapitools.client.apis.*
+import org.openapitools.client.models.*
+import org.vitrivr.engine.core.model.content.element.AudioContent
+import org.vitrivr.engine.core.util.extension.toDataURL
+import java.awt.image.BufferedImage
+
+data class JobResult<S>(
+        val status: JobState,
+        val result: S?
+)
+
+class JobWrapper<T, S>(
+        private val startJobFunc: (T) -> JobStatus,
+        private val getJobResultFunc: (String) -> JobResult<S>
+) {
+
+    fun executeJob(inp: T): S {
+        val jobStatus = startJobFunc(inp)
+        var jobResult = getJobResultFunc(jobStatus.id)
+
+        while (jobResult.status != JobState.complete) {
+            if (jobResult.status == JobState.failed) {
+                throw Exception("Job failed.")
+            }
+            // Wait for a while before checking the job status again
+            Thread.sleep(1000)
```

Comment:
I think this should at least be a configurable polling and have some more logic on how to handle fails / no results etc.